apps/progs.pl

```diff
@@ -143,8 +178,53 @@
 
     my %cipher_disabler = (
         des3  => "des",
-        desx  => "des",
-        cast5 => "cast",
+    );
+    my %cipher_deprecated = (
+        "des" =>         [ "3_0", "des" ],
+        "idea" =>        [ "3_0", "idea" ],
+        "seed" =>        [ "3_0", "seed" ],
+        "rc4" =>         [ "3_0", "rc4" ],
+        "rc4-40" =>      [ "3_0", "rc4" ],
+        "rc2" =>         [ "3_0", "rc2" ],
+        "bf" =>          [ "3_0", "bf" ],
+        "cast" =>        [ "3_0", "cast" ],
+        "rc5" =>         [ "3_0", "rc5" ],
+        "idea-cbc" =>    [ "3_0", "idea" ],
+        "idea-ecb" =>    [ "3_0", "idea" ],
+        "idea-cfb" =>    [ "3_0", "idea" ],
+        "idea-ofb" =>    [ "3_0", "idea" ],
+        "seed-cbc" =>    [ "3_0", "seed" ],
+        "seed-ecb" =>    [ "3_0", "seed" ],
+        "seed-cfb" =>    [ "3_0", "seed" ],
+        "seed-ofb" =>    [ "3_0", "seed" ],
+        "rc2-cbc" =>     [ "3_0", "rc2" ],
+        "rc2-ecb" =>     [ "3_0", "rc2" ],
+        "rc2-cfb" =>     [ "3_0", "rc2" ],
+        "rc2-ofb" =>     [ "3_0", "rc2" ],
+        "rc2-64-cbc" =>  [ "3_0", "rc2" ],
+        "rc2-40-cbc" =>  [ "3_0", "rc2" ],
+        "bf-cbc" =>      [ "3_0", "bf" ],
+        "bf-ecb" =>      [ "3_0", "bf" ],
+        "bf-cfb" =>      [ "3_0", "bf" ],
+        "bf-ofb" =>      [ "3_0", "bf" ],
+        "cast5-cbc" =>   [ "3_0", "cast" ],
+        "cast5-ecb" =>   [ "3_0", "cast" ],
+        "cast5-cfb" =>   [ "3_0", "cast" ],
+        "cast5-ofb" =>   [ "3_0", "cast" ],
+        "cast-cbc" =>    [ "3_0", "cast" ],
+        "rc5-cbc" =>     [ "3_0", "rc5" ],
+        "rc5-ecb" =>     [ "3_0", "rc5" ],
+        "rc5-cfb" =>     [ "3_0", "rc5" ],
+        "rc5-ofb" =>     [ "3_0", "rc5" ],
+        "des-ecb" =>     [ "3_0", "des" ],
+        "des-ede" =>     [ "3_0", "des" ],
+        "des-cbc" =>     [ "3_0", "des" ],
+        "des-ede-cbc" => [ "3_0", "des" ],
+        "des-cfb" =>     [ "3_0", "des" ],
+        "des-ede-cfb" => [ "3_0", "des" ],
+        "des-ofb" =>     [ "3_0", "des" ],
+        "des-ede-ofb" => [ "3_0", "des" ],
+        "desx" =>        [ "3_0", "des" ],
     );
```

Comment:
It turn out that these deprecations are premature too.

Reply:
Okay, should I also remove the Perl changes associated with these or leave them for future use?

Reply:
Hmmm, just so we don't undo the framework you've worked out, I would keep the empty hash tables. Possibly with a comment.

Reply:
Okay, I've done this.

Reply:
I  actually think this should all be removed. This is now dead code. If it gets resurrected any time soon then I'm sure you'll keep a copy locally. Otherwise it will just stick around in the code not doing anything.

Reply:
It wasn't difficult to write even for a Perl-denier, so I'm easy either way.

Reply:
I've pushed a fixup with this removed.  I can unroll it if that's the consensus.